packages/platform/types/bookings/2024-08-13/inputs/update-location.input.ts

```diff
@@ -0,0 +1,135 @@
+import { ApiPropertyOptional, getSchemaPath } from "@nestjs/swagger";
+import { ApiProperty as DocsProperty } from "@nestjs/swagger";
+import { plainToInstance } from "class-transformer";
+import { IsOptional } from "class-validator";
+import { IsString, MinLength, IsUrl, IsPhoneNumber, IsDefined } from "class-validator";
+import type { ValidationOptions, ValidatorConstraintInterface } from "class-validator";
+import { registerDecorator, validate, ValidatorConstraint } from "class-validator";
+
+import {
+  BookingInputAddressLocation_2024_08_13,
+  BookingInputAttendeeAddressLocation_2024_08_13,
+  BookingInputAttendeeDefinedLocation_2024_08_13,
+  BookingInputAttendeePhoneLocation_2024_08_13,
+  BookingInputLinkLocation_2024_08_13,
+  BookingInputPhoneLocation_2024_08_13,
+} from "./location.input";
+
+export class UpdateInputAddressLocation_2024_08_13 extends BookingInputAddressLocation_2024_08_13 {
+  @IsDefined()
+  @IsString()
+  @MinLength(1)
+  @DocsProperty({ example: "123 Example St, City, Country" })
+  address!: string;
+}
+
+export class UpdateBookingInputAttendeeAddressLocation_2024_08_13 extends BookingInputAttendeeAddressLocation_2024_08_13 {}
+
+export class UpdateBookingInputAttendeeDefinedLocation_2024_08_13 extends BookingInputAttendeeDefinedLocation_2024_08_13 {}
+
+export class UpdateBookingInputAttendeePhoneLocation_2024_08_13 extends BookingInputAttendeePhoneLocation_2024_08_13 {}
+
+export class UpdateBookingInputLinkLocation_2024_08_13 extends BookingInputLinkLocation_2024_08_13 {
+  @IsDefined()
+  @IsUrl()
+  @DocsProperty({ example: "https://meet.google.com/txi-fein-xyz" })
+  link!: string;
+}
+
+export class UpdateBookingInputPhoneLocation_2024_08_13 extends BookingInputPhoneLocation_2024_08_13 {
+  @IsDefined()
+  @IsPhoneNumber()
+  @DocsProperty({ example: "+37120993151" })
+  phone!: string;
+}
+
+export type UpdateBookingInputLocation_2024_08_13 =
+  | UpdateInputAddressLocation_2024_08_13
+  | UpdateBookingInputAttendeeAddressLocation_2024_08_13
+  | UpdateBookingInputAttendeeDefinedLocation_2024_08_13
+  | UpdateBookingInputAttendeePhoneLocation_2024_08_13
+  | UpdateBookingInputLinkLocation_2024_08_13
+  | UpdateBookingInputPhoneLocation_2024_08_13;
+
+@ValidatorConstraint({ async: true })
+class UpdateBookingInputLocationValidator_2024_08_13 implements ValidatorConstraintInterface {
+  private validationMessage = "";
+
+  private classTypeMap: { [key: string]: new () => UpdateBookingInputLocation_2024_08_13 } = {
+    address: UpdateInputAddressLocation_2024_08_13,
+    attendeeAddress: UpdateBookingInputAttendeeAddressLocation_2024_08_13,
+    attendeeDefined: UpdateBookingInputAttendeeDefinedLocation_2024_08_13,
+    attendeePhone: UpdateBookingInputAttendeePhoneLocation_2024_08_13,
+    link: UpdateBookingInputLinkLocation_2024_08_13,
+    phone: UpdateBookingInputPhoneLocation_2024_08_13,
+  };
+
+  async validate(location: { type: string } | string) {
+    if (typeof location === "string") {
```

Comment:
We can remove `if (typeof location === "string") {` because in `packages/platform/types/bookings/2024-08-13/inputs/location.input.ts` we have it because people could at first pass `booking.location` as a string and then we added specific objects aka `{type: phone, phone: x}` etc.

But since this is a new endpoint we only accept the objects from the start so its fine - we don't accept string location anymore. Does it make sense?